File: tournaments/numbersGrouping/numbersGrouping.cs
int numbersGrouping(int[] a) {
    var result = 0;
    var store = new int[1000000];

    for (var i = 0; a.Length > i; ++i) {
        var gr = getGroup(a[i]);
        if (0 == store[gr]) ++store[gr];
    }

    for (var i = 0; store.Length > i; ++i) {
        result += store[i];
    }

    return result + a.Length;
}

int getGroup(int n) {
    var result = (n - 1) / 10000;

    return result;
}

File: tournaments/numbersGrouping/numbersGrouping.js
function numbersGrouping(a) {
    const getGroup = function(a) {
        return ~~((a - 1) / 10000);
    };

    const n = a.length;
    const store = Array(1000000).fill(0);
    let result = 0;

    for (const item of a) {
        const gr = getGroup(item);

        if (!store[gr]) {
            ++store[gr];
            ++result;
        }
    }

    return result + n;
}
